discovery-provider/src/tasks/index_aggregate_user.py

```diff
@@ -0,0 +1,404 @@
+import logging
+import time
+import sqlalchemy as sa
+from src.tasks.celery_app import celery
+from src.utils.redis_constants import most_recent_indexed_aggregate_user_block_redis_key
+from src.tasks.calculate_trending_challenges import get_latest_blocknumber
+
+logger = logging.getLogger(__name__)
+
+# Names of the aggregate tables to update
+AGGREGATE_USER = "aggregate_user"
+DEFAULT_UPDATE_TIMEOUT = 60
```

Comment:
60s seems too low for a redis lock

Reply:
i think we should revisit these other values then. the problem i'm thinking about is after 60sec the lock is released another celery task comes up and starts running this query again even though there's one already up. we've seen some of these queries take like up to 10 mins right?

Reply:
Yeah so we could have a buildup of inserts that are blocked. Yeah previously we've seen around 10 minutes but haven't seen any that long for this upsert. I'll increase it to 5 minutes. aggregate_track and aggregate_playlist take less than 10 seconds so maybe they're OK.

Reply:
I would actually set the lock expiry longer than we think it'll take since we don't incur any penalty. When the task is done it should clear the lock anyway. The inverse can't be said if we release too early. Then multiple different tasks could come up and try to run this in parallel. 

If there is a buildup i wonder if we can try to find a way to wipe the backlog of jobs.